locations/spiders/habit_burger.py

```diff
@@ -1,80 +1,32 @@
+
+class HabitBurgerSpider(SitemapSpider, StructuredDataSpider):
     name = "habit-burger"
     allowed_domains = ["habitburger.com"]
-    start_urls = ("https://www.habitburger.com/locations/all/",)
-
-    def parse(self, response):
-        regions = response.xpath("//h2").xpath("@id").extract()
-        non_us_cities = ["phnom penh", "shanghai", "hangzhou"]
-
-        for region in regions:
-            region_body = response.xpath(
-                '//h2[@id = "{region}"]/following-sibling::ul[@class = "reglist"]'.format(region=region)
-            ).get()
-            pois = Selector(text=region_body).css(".loc").extract()
-
-            for poi in pois:
-                ref = Selector(text=poi).xpath("//a/@href").extract()[0].split("/")[-1]
-                name = Selector(text=poi).xpath("//h4/text()").extract()
-                if name == []:
-                    name = Selector(text=poi).xpath("//h3/text()").extract()
-                name = "".join(name)
-
-                map_link = Selector(text=poi).xpath('//div[@class = "locaddress"]/a').xpath("@href").extract_first()
-                lat, long = None, None
-                if "daddr" in map_link:
-                    coords = map_link.split("daddr=")[1].split(",")
-                    lat = coords[0]
-                    long = coords[1]
-
-                addr = Selector(text=poi).xpath('//div[@class = "locaddress"]/a').extract_first()
-                addr = Selector(text=addr).xpath("//a/text()").extract()
-                addr = [a.strip() for a in addr]
-
-                addr_full = ", ".join(addr)
-                street = ", ".join(addr[:-1])
-                city, state, postcode = None, None, None
-
-                if region in ["cambodia", "china"]:
-                    for c in non_us_cities:
-                        if c in poi.lower():
-                            city = c.capitalize()
-                    country = region.capitalize()
-                else:
-                    city = addr[-1].split(", ")[0]
-                    state_postcode = addr[-1].split(", ")[1].split(" ")
-                    if len(state_postcode) > 1:
-                        state = state_postcode[0]
-                        postcode = state_postcode[1]
-                    country = "US"
-
-                phone = Selector(text=poi).xpath('//div[@class="locinfo"]/text()').get()
-                phone = phone.strip() if phone else None
-                opening_hours = Selector(text=poi).xpath('//div[@class="lochrs"]/text()').extract()
-                opening_hours = opening_hours = (
-                    ", ".join([hours.strip() for hours in opening_hours]) if opening_hours else None
+    sitemap_urls = ["https://www.habitburger.com/locations-sitemap.xml"]
+    sitemap_rules = [("", "parse_sd")]
+    wanted_types = ["Restaurant"]
+
+    def inspect_item(self, item, response):
+        openingHour = json.loads(response.xpath('//*[@id="content"]/script[1]/text()').get())
```

Comment:
This is to get the ld json? If so doing it in `pre_process_data` or `post_process_item` will already have `ld_data` available.

Reply:
I've not checked, but the first example I looked at was valid ld format, but with a none default time format, we need a better solution, but for now the following may work:
`oh.from_linked_data(ld_data, time_format="%I:%M%p")`

Reply:
@Cj-Malone  
You need an update here, you want me to

```
        elif linked_data.get("openingHours"):
            rules = linked_data["openingHours"]
            if not isinstance(rules, list):
                rules = re.findall(
                    r"((\w{2,3}|\w{2,3}\s?\-\s?\w{2,3}|(\w{2,3},)+\w{2,3})\s(\d\d:\d\d)\s?\-\s?(\d\d:\d\d))",
                    rules,
                )
                rules = [r[0] for r in rules]

            for rule in rules:
                days, time_ranges = rule.split(" ", 1)

                if time_ranges.lower() == "closed":
                    continue

                for time_range in time_ranges.split(","):
                    start_time, end_time = time_range.split("-")

                    start_time = start_time.strip()
                    end_time = end_time.strip()

                    if "-" in days:
                        start_day, end_day = days.split("-")

                        start_day = sanitise_day(start_day)
                        end_day = sanitise_day(end_day)

                        for day in day_range(start_day, end_day):
                            self.add_range(day, start_time, end_time, time_format)
                    else:
                        for day in days.split(","):
                            self.add_range(day, start_time, end_time, time_format)
```